docs/source/cuda/random.rst

```diff
@@ -64,3 +64,27 @@ Here is a sample program that uses the random number generator::
 
     compute_pi[blocks, threads_per_block](rng_states, 10000, out)
     print('pi:', out.mean())
+
+An example managing RNG state size and using a 3D grid
+''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+The size of the RNG states scales with the number of threads using the RNG, so
+it is often better to use strided loops in conjunction with the RNG in order to
+keep the state size manageable.
+
+In the following example that initializes a large 3D array with random numbers,
```

Comment:
```suggestion
In the following example, that initializes a large 3D array with random numbers,
```

Reply:
This suggestion feels like it's in the grammatical no-man's land between ", which", and "that" - c.f. the examples on slide 11 of https://cims.nyu.edu/~donev/Teaching/WrittenOral/Lecture-Language.pdf:

- “Consider the Pei matrix, which is positive definite.”
- “Consider a Pei matrix that is positive definite.”

Perhaps

```suggestion
In the following example, which initializes a large 3D array with random numbers,
```

is closer to being clear? (though, I still prefer my original "that").

Reply:
I prefer the latter, but am not going to quibble :)

Reply:
by "the latter", do you mean ", which"?

Reply:
yes :) but really am not going to quibble.